framework/src/meshlet_helpers.cpp

```diff
@@ -0,0 +1,139 @@
+					// append to meshlets
+					meshlets.insert(std::end(meshlets), std::begin(tmpMeshlets), std::end(tmpMeshlets));
+				}
+			}
+		}
+		return meshlets;
+	}
+
+	template <typename F>
+	std::vector<meshlet> divide_into_meshlets_cached(gvk::serializer& aSerializer,
+		const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModels,
+		F&& aMeshletDivision, const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		std::vector<meshlet> meshlets;
+		if (aSerializer.mode() == gvk::serializer::mode::serialize) {
+			meshlets = divide_into_meshlets(aModels, aMeshletDivision, aMaxVertices, aMaxIndices, aCombineSubmeshes);
+		}
+		aSerializer.archive(meshlets);
+
+		if (aSerializer.mode() == serializer::mode::deserialize) {
+			std::unordered_map<std::string, gvk::model> modelLookup;
+			for (auto& pair : aModels) {
+				auto model = std::get<avk::resource_ownership<gvk::model_t>>(pair);
+				modelLookup.insert_or_assign(model->path(), model.own());
+			}
+			for (auto& meshlet : meshlets) {
+				meshlet.mModel = modelLookup.at(meshlet.mModelPath);
+			}
+		}
+
+		return meshlets;
+	}
+
+	template <typename F>
+	std::vector<meshlet> divide_vertices_into_meshlets(const std::vector<glm::vec3>& aVertices,
+		const std::vector<uint32_t>& aIndices, avk::resource_ownership<gvk::model_t> aModel,
+		const std::optional<mesh_index_t> aMeshIndex, const uint32_t aMaxVertices, const uint32_t aMaxIndices,
+		F&& aMeshletDivision)
+	{
+		std::vector<meshlet> tmpMeshlets;
+		if constexpr (std::is_assignable_v<std::function<std::vector<meshlet>(const std::vector<uint32_t>&tIndices, avk::resource_ownership<model_t> tModel, const std::optional<mesh_index_t> tMeshIndex, const uint32_t tMaxVertices, const uint32_t tMaxIndices)>, decltype(aMeshletDivision)>) {
+			tmpMeshlets = aMeshletDivision(aIndices, aModel, aMeshIndex, aMaxVertices, aMaxIndices);
+		}
+		else if constexpr (std::is_assignable_v<std::function<std::vector<meshlet>(const std::vector<glm::vec3> &tVertices, const std::vector<uint32_t>&tIndices, avk::resource_ownership<model_t> tModel, const std::optional<mesh_index_t> tMeshIndex, const uint32_t tMaxVertices, const uint32_t tMaxIndices)>, decltype(aMeshletDivision)>) {
+			tmpMeshlets = aMeshletDivision(aVertices, aIndices, aModel, aMeshIndex, aMaxVertices, aMaxIndices);
+		}
+		else {
+#if defined(_MSC_VER) && defined(__cplusplus)
+			static_assert(false);
+#else
+			assert(false);
+#endif
+			throw avk::logic_error("No lambda has been passed to divide_into_meshlets.");
+		}
+		return tmpMeshlets;
+	}
+
+	std::vector<meshlet> divide_into_very_bad_meshlets(const std::vector<uint32_t>& aIndices,
+		avk::resource_ownership<model_t> aModel,
+		const std::optional<mesh_index_t> aMeshIndex,
+		const uint32_t aMaxVertices, const uint32_t aMaxIndices)
```

Comment:
```suggestion
	std::vector<meshlet> divide_into_very_bad_meshlets(const std::vector<uint32_t>& aIndices,
		const model_t& aModel,
		std::optional<mesh_index_t> aMeshIndex,
		uint32_t aMaxVertices, uint32_t aMaxIndices)
```

Adapt according to the requested changes described in `divide_vertices_into_meshlet`!